pkg/trace/traceutil/span_test.go

```diff
@@ -146,6 +146,66 @@ func TestTopLevelWithTag(t *testing.T) {
 	assert.Equal(float64(42), tr[1].Metrics["custom"], "custom metric should still be here")
 }
 
+func TestTopLevelTypicalOTLP(t *testing.T) {
```

Comment:
Can these tests be refactored together to reduce the duplication here? Or at least make them one test with 3 sub tests?